libs/langchain/langchain/retrievers/google_cloud_docai_warehouse.py

```diff
@@ -0,0 +1,118 @@
+"""Retriever wrapper for Google Cloud DocAI Warehouse on Gen App Builder."""
+from typing import TYPE_CHECKING, Any, Dict, List, Optional
+
+from langchain.callbacks.manager import CallbackManagerForRetrieverRun
+from langchain.docstore.document import Document
+from langchain.pydantic_v1 import root_validator
+from langchain.schema import BaseRetriever
+from langchain.utils import get_from_dict_or_env
+
+if TYPE_CHECKING:
+    from google.cloud.contentwarehouse_v1 import (
+        DocumentServiceClient,
+        RequestMetadata,
+        SearchDocumentsRequest,
+    )
+    from google.cloud.contentwarehouse_v1.services.document_service.pagers import (
+        SearchDocumentsPager,
+    )
+
+
+class GoogleDocaiWarehouseSearchRetriever(BaseRetriever):
+    """A retriever based on DocAI Warehouse.
+
+    Documents should be created and documents should be uploaded
+        in a separate flow, and this retriever uses only DocAI
+        schema_id provided to search for revelant documents.
+
+    More info: https://cloud.google.com/document-ai-warehouse.
+    """
+
+    location: str = "us"
+    "GCP location where DocAI Warehouse is placed."
+    project_id: str
+    "GCP project number, should contain digits only."
+    schema_id: Optional[str] = None
+    "DocAI Warehouse schema to queary against. If nothing is provided, all documents "
+    "in the project will be searched."
```

Comment:
The most important part of this comment is changing the parameter `project_id` to `project_number`. Since "Project ID" refers to a different identifier in Google Cloud.